src/Microsoft.ML.Transforms/Text/StopWordsRemoverTransform.cs

```diff
@@ -857,101 +894,157 @@ private CustomStopWordsRemoverTransform(IHost host, ModelLoadContext ctx, IDataV
         }
 
-        public static CustomStopWordsRemoverTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        // Factory method for SignatureLoadModel.
+        private static CustomStopWordsRemoverTransform Create(IHostEnvironment env, ModelLoadContext ctx)
         {
             Contracts.CheckValue(env, nameof(env));
-            env.CheckValue(ctx, nameof(ctx));
+            var host = env.Register(RegistrationName);
+            host.CheckValue(ctx, nameof(ctx));
             ctx.CheckAtModel(GetVersionInfo());
-            env.CheckValue(env, nameof(env));
+            return new CustomStopWordsRemoverTransform(host, ctx);
+        }
+
+        // Factory method for SignatureDataTransform.
+        internal static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(args, nameof(args));
             env.CheckValue(input, nameof(input));
 
-            var h = env.Register(RegistrationName);
-            return h.Apply("Loading Model", ch => new CustomStopWordsRemoverTransform(h, ctx, input));
+            env.CheckValue(args.Column, nameof(args.Column));
+            var cols = new (string input, string output)[args.Column.Length];
+            for (int i = 0; i < cols.Length; i++)
+            {
+                var item = args.Column[i];
+                cols[i] = (item.Source ?? item.Name, item.Name);
+            }
+            //IVAN propagate other options of argument class.
+            return new CustomStopWordsRemoverTransform(env, args.Stopword, cols).MakeDataTransform(input);
         }
 
-        public override void Save(ModelSaveContext ctx)
+        // Factory method for SignatureLoadDataTransform.
+        private static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+            => Create(env, ctx).MakeDataTransform(input);
+
+        // Factory method for SignatureLoadRowMapper.
+        private static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, ISchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        protected override IRowMapper MakeRowMapper(ISchema schema) => new Mapper(this, Schema.Create(schema));
+
+        private sealed class Mapper : MapperBase
         {
-            Host.CheckValue(ctx, nameof(ctx));
-            ctx.CheckAtModel();
-            ctx.SetVersionInfo(GetVersionInfo());
+            private readonly ColumnType[] _types;
+            private readonly CustomStopWordsRemoverTransform _parent;
 
-            // *** Binary format ***
-            // <base>
-            SaveBase(ctx);
+            public Mapper(CustomStopWordsRemoverTransform parent, Schema inputSchema)
+             : base(parent.Host.Register(nameof(Mapper)), parent, inputSchema)
+            {
+                _parent = parent;
+                _types = new ColumnType[_parent.ColumnPairs.Length];
```

Comment:
>_types [](start = 16, length = 6)

You might want to fill this array

Reply:
Why are _types needed? Isnt the output always text?

---
In reply to: [227894352](https://github.com/dotnet/machinelearning/pull/1338#discussion_r227894352) [](ancestors = 227894352)